### PR TITLE
Route and JS pathway for the new teacher homepage 

### PR DESCRIPTION
**Pull request by Erin007**
This PR sets up the route (studio.code.org/teacher) and a JS entry point in preparation for the new  teacher homepage. To test the JS entry point, I render a "test" React component. 

initial new teacher homepage

This work incrementally builds toward one large React component on the teacher homepage whose children are currently a work in progress.

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @tanyaparker @joshlory

File: apps/Gruntfile.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ testsContext.keys().forEach(testsContext);
   var appsToBuild = SINGLE_APP ? [SINGLE_APP] : ALL_APPS;
 
   var ace_suffix = envConstants.DEV ? '' : '-min';
-  var dotMinIfNotDev = envConstants.DEV ? '' : '.min';
+  // var dotMinIfNotDev = envConstants.DEV ? '' : '.min'; <---- linter threw an error that this variable is defined, but never used. Should I delete it?
```
Comment (breville) [inline apps/Gruntfile.js]: I saw the same and am about to delete it when I merge https://github.com/code-dot-org/code-dot-org/pull/14229

File: apps/Gruntfile.js [hunk 2]
```diff
@@ -429,7 +429,8 @@ testsContext.keys().forEach(testsContext);
     'raceInterstitial':             './src/sites/studio/pages/raceInterstitial.js',
     'layouts/_terms_interstitial':  './src/sites/studio/pages/layouts/_terms_interstitial.js',
     'maker/setup':                  './src/sites/studio/pages/maker/setup.js',
-    'scriptOverview':               './src/sites/studio/pages/scriptOverview.js'
+    'scriptOverview':               './src/sites/studio/pages/scriptOverview.js',
+    'home/new_teacher_homepage':         './src/sites/studio/pages/home/new_teacher_homepage.js'
```
Comment (breville) [inline apps/Gruntfile.js]: nit: can just indent to align with above

Comment (breville) [inline apps/Gruntfile.js]: for filenames that aren't user-facing, and don't have obvious confusion with an existing page, I think it's okay to just name them as they will be named long-term, e.g. `home/teacher_homepage`.

Comment (breville) [reply apps/Gruntfile.js]: Also does it make sense to create `pages/home/` or is it okay to just put in `pages/'?

**Pull request merged**